src/main/java/org/folio/service/impl/RequestServiceImpl.java

```diff
@@ -67,7 +67,12 @@ public RequestWrapper createSecondaryRequest(Request request, String borrowingTe
         return executionService.executeSystemUserScoped(lendingTenantId, () -> {
           log.info("createSecondaryRequest:: creating requester {} in lending tenant ({})",
             requesterId, lendingTenantId);
-          userCloningService.clone(primaryRequestRequester);
+          User shadowUser = userCloningService.clone(primaryRequestRequester);
+
+          if(!primaryRequestRequester.getPatronGroup().equals(shadowUser.getPatronGroup())) {
```

Comment:
Could it be possible for getPatronGroup() to return a null value?

Reply:
Yes, I did not find that patron group is required, so added null check